core/tee/tee_rpmb_fs.c

```diff
@@ -3173,3 +3174,64 @@ bool __weak plat_rpmb_key_is_ready(void)
 {
 	return true;
 }
+
+#ifdef CFG_WITH_STATS
+TEE_Result rpmb_mem_stats(struct pta_stats_alloc *stats, bool reset)
+{
+	TEE_Result res = TEE_ERROR_GENERIC;
+	struct rpmb_fat_entry *fe = NULL;
+	tee_mm_entry_t *mm = NULL;
+	tee_mm_pool_t pool = { };
+	bool pool_result = false;
+	paddr_size_t pool_sz = 0;
+
+	if (!fs_par)
+		return TEE_ERROR_STORAGE_NOT_AVAILABLE;
+
+	pool_sz = fs_par->max_rpmb_address - RPMB_STORAGE_START_ADDRESS;
+	pool_result = tee_mm_init(&pool, RPMB_STORAGE_START_ADDRESS,
+				  pool_sz, RPMB_BLOCK_SIZE_SHIFT,
+				  TEE_MM_POOL_HI_ALLOC);
+	if (!pool_result)
+		return TEE_ERROR_OUT_OF_MEMORY;
+
+	mutex_lock(&rpmb_mutex);
+	res = fat_entry_dir_init();
+	if (res)
+		goto out;
+
+	/*
+	 * The pool is used to represent the current RPMB layout. To find
+	 * a slot for the file tee_mm_alloc is called on the pool. Thus
+	 * if it is not NULL the entire FAT must be traversed to fill in
+	 * the pool.
+	 */
+	while (true) {
+		uint32_t fat_address = 0;
+
+		res = fat_entry_dir_get_next(&fe, &fat_address);
+		if (res || !fe)
+			break;
+
+		if (!(fe->flags & FILE_IS_ACTIVE) || !fe->data_size)
+			continue;
+
+		mm = tee_mm_alloc2(&pool, fe->start_address, fe->data_size);
+		if (!mm) {
+			res = TEE_ERROR_GENERIC;
+			break;
+		}
+	}
+
+out:
+	fat_entry_dir_deinit();
+	mutex_unlock(&rpmb_mutex);
+
+	if (res)
+		return res;
+
+	tee_mm_get_pool_stats(&pool, stats, reset);
```

Comment:
Shouldn't the pool be destroyed after this point?

Reply:
Indeed, thanks.